Jint/Native/Array/ArrayInstance.cs

```diff
@@ -210,65 +280,69 @@ public override bool DefineOwnProperty(string propertyName, PropertyDescriptor d
 
         public uint GetLength()
         {
-            return TypeConverter.ToUint32(_length.Value);
+            return GetLengthValue();
         }
 
         public override IEnumerable<KeyValuePair<string, PropertyDescriptor>> GetOwnProperties()
         {
-            foreach(var entry in _array)
+            if (_dense != null)
             {
-                yield return new KeyValuePair<string, PropertyDescriptor>(TypeConverter.ToString(entry.Key), entry.Value);
+                for (var i = 0; i < _dense.Length; i++)
+                {
+                    if (_dense[i] != null)
```

Comment:
If it's dense, shouldn't the first null value mark that all subsequent ones are not used and break the loop?

Reply:
It's expected to be dense, but allows sparseness as currently shares the ArrayProtype API. So you could call delete in the middle if I remember right. Ideally we could natively support unshift with moving start index for array. But this requires some work and moving operations closer to arrayinstance.